src/engine/board.ts

```diff
@@ -34,9 +35,46 @@ export default class Board {
     public movePiece(fromSquare: Square, toSquare: Square) {
         const movingPiece = this.getPiece(fromSquare);        
         if (!!movingPiece && movingPiece.player === this.currentPlayer) {
+            if (movingPiece instanceof Pawn)
+            {
+                let nr1=2;
+                let nr2=-2;
+                if (toSquare.row-this.findPiece(movingPiece).row==nr1 || (toSquare.row-this.findPiece(movingPiece).row)==nr2)
+                {
+                    movingPiece.possibleEnPassant=true;
+                }
+            }
             this.setPiece(toSquare, movingPiece);
             this.setPiece(fromSquare, undefined);
+            let squareBehind:Square;
+            if (this.currentPlayer===Player.WHITE)
+                squareBehind=new Square(toSquare.row-1,toSquare.col);
+            else
+                squareBehind=new Square(toSquare.row+1,toSquare.col)
```

Comment:
square behind poate fi de asemenea const daca il initializam din prima:
`const squareBehind = this.currentPlayer === Player.WHITE ? new Square(toSquare.row-1,toSquare.col) : new Square(toSquare.row+1,toSquare.col)`

Reply:
solved